REPOSITORY-GUIDELINES.md

```diff
@@ -20,13 +20,24 @@ Core repositories are all those located under the
 
 ### Core Repository Requirements
 
-- Repository must live under github.com/knative/project-name
-- Must adopt the Knative Code of Conduct
+- Repository must live under github.com/knative/project-name.
+- Must adopt the Knative Code of Conduct.
 - All code projects must use the Apache License version 2.0.
 - Documentation repositories must use Creative Commons License version 4.0.
 - All OWNERS must be members of the Knative community.
 - Repository creation must be approved by the Technical Oversight Committee.
 
+### Requirements for Repository Donation
+
+The Technical Oversight Committee may allow repositories to be donated to the
+Knative project. In *addition* to meeting the core repository requirements,
+donated repositories must:
+
+- Adopt the Google CLA bot.
+- All contributors to the donated repository must have signed the Google CLA.
+- Boilerplate text across all files should attribute copyright as follows:
+  `"Copyright <year> The Knative Authors"

```

Comment:
I'm not a lawyer, but: would this replace existing copyright text, or would it augment existing copyright text?

```
    Copyright 1998-2019 ExampleCo International LLC 
```

Does that become

```
    Copyright 1998-2019 ExampleCo International LLC 
    Copyright 2019-current year The Knative Authors
```

Or does it ditch the `ExampleCo` attribution (assuming ExampleCo is willing and able to transfer copyright)?

Reply:
I would vote to replace the copyright with the new one and avoid any shinanigans with dual copyright and copy left in the are litigations in the future.
Donating means donating everything including the copyright.

Attribution and credit, still there to individual contributors via git history/commits
Maybe clarify that the git repo should be donated with history in place.

Reply:
+1 to replacing the copyright, that's driven by the CLA as well -- the copyright is reassigned to Google / Knative authors. We shouldn't have dual copyright notices on the files.

If/when we can't have all contributors of the code sign the CLA, then we'll need to work through the details, but as long as we can meet that initial requirement this should be OK.

Reply:
All eight of the contributors have signed the CLA, fyi

Reply:
Re timing, is it ok to transfer first, and then add the copyright text, or must that happen prior to the transfer?

Reply:
I asked about this. Basically, it's not super important if it's one attribution or two, copyright arises from the particular facts of contributors and contributions rather than the copyright header. 

So I think that (1) we switch it to "The Knative Authors" and (2) we can make the change either before or after moving the repository, whichever's easiest.

Reply:
That's my understanding as well.

Reply:
So, is there a request to change the text here?

Reply:
Yes, do change the text to "The Knative Authors".

But two points to note:
1. copyright is not being assigned to anyone in this scenario.  Google's CLA is not a copyright assignment, it's a license (hence the name, Contributor License Agreement).  The original authors still retain copyright to the code they wrote.  When we say "The [Project] Authors", that still includes the original copyright holders, we're just using a broader term to identify them.

2. The reason we use this phrasing is for the benefit of downstream users who redistribute the code.  The Apache license requires that anyone who redistributes the code "retain [...] all copyright, patent, trademark, and attribution notices" (Section 4c).  If every file in the project had different copyright statements, then users would be obligated to find and reproduce all of them anytime they distribute the project.  We don't want them to have to do that, so we have simple, consistent copyright statements on all the files which makes it easier for them to fulfill their notice obligations.

Reply:
Thanks Will. I've revised my comment above according. Sounds like we should be good to proceed with the current guidance proposed by @pmorie.